### DISCUSSION TITLE
MAKE Pair-allegro error with the newer LAMMPS version

### DISCUSSION THREAD

**Discussion by upavelru** [mir-group/pair_nequip_allegro#32] [Q&A]
Hi,

I need to run some Monte Carlo simulation with the developed Allegro potential. However, Monte Carlo was implemented only in the newer LAMMPS-22Dec-2022 version (not LAMMPS-stable_29Sep2021_update2). I've tried to compile this version with the Allegro pair style but I've got this error:
Building CXX object CMakeFiles/lammps.dir/home/path/to/lammps-22Dec2022/src/pair_allegro.cpp.o
/home/path/to/lammps-22Dec2022/src/pair_allegro.cpp: In member function ‘virtual void LAMMPS_NS::PairAllegro::init_style()’:
/home/path/to/lammps-22Dec2022/src/pair_allegro.cpp:125:33: error: ‘int LAMMPS_NS::NeighRequest::half’ is protected within this context
125 | neighbor->requests[irequest]->half = 0;
| ^~~~
In file included from /home/path/to/lammps-22Dec2022/src/pair_allegro.cpp:26:
/home/path/to/lammps-22Dec2022/src/neigh_request.h:54:7: note: declared protected here
54 | int half; // half neigh list (set by default)
| ^~~~
/home/path/to/lammps-22Dec2022/src/pair_allegro.cpp:126:33: error: ‘int LAMMPS_NS::NeighRequest::full’ is protected within this context
126 | neighbor->requests[irequest]->full = 1;
| ^~~~
In file included from /home/path/to/lammps-22Dec2022/src/pair_allegro.cpp:26:
/home/path/to/lammps-22Dec2022/src/neigh_request.h:55:7: note: declared protected here
55 | int full; // full neigh list
| ^~~~
/home/path/to/lammps-22Dec2022/src/pair_allegro.cpp:128:33: error: ‘int LAMMPS_NS::NeighRequest::ghost’ is protected within this context
128 | neighbor->requests[irequest]->ghost = 1;
| ^~~~~
In file included from /home/path/to/lammps-22Dec2022/src/pair_allegro.cpp:26:
/home/path/to/lammps-22Dec2022/src/neigh_request.h:68:7: note: declared protected here
68 | int ghost; // 1 if includes ghost atom neighbors
| ^~~~~

make[2]: *** [CMakeFiles/lammps.dir/build.make:4752: CMakeFiles/lammps.dir/home/path/to/lammps-22Dec2022/src/pair_allegro.cpp.o] Error 1

There is no compilation error with the recommended version LAMMPS-stable_29Sep2021_update2, but Monte Carlo was not implemented in it, so I need to use the newer version.

Is it possible to fix this problem?

Thank you!

**Comment by Linux-cpp-lisp** [comment]
Hi @upavelru ,

Thanks for your interest in our work!

Use instead `pair_allegro` for example [here](https://github.com/mir-group/pair_allegro/tree/multicut) which should support the newest versions.
